### PR TITLE
Add TabView to Autohide Layout

### PR DESCRIPTION
**Pull request by Rohitth007**
**What does this PR do?**  
* This commit adds a closed tab view of width TAB_WIDTH (3) for each of the side
panels when in the autohide layout. The initial layout when the app
is started is left panel open right panel closed. This improves the UX and also makes it easier
for first time users to figure out what is happening.

[#zulip-terminal > Autohide Tabs](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/Autohide.20Tabs)
**Tested?** 
- [x] Manually
- [x] Existing tests (adapted, if necessary)
- [x] New tests added (for any new behavior)
- [x] Passed linting & tests (each commit)

**Commit flow** 

* The first commit adds the TabView widget class
and the second commit uses it in autohide layout.

**Notes & Questions** 

* The title for the left tab is named as INBOX currently for lack of something more apt.

**Interactions** 

- Blocks finalization of #1074 

**Visual changes** 
OnStartup

BothPanelsClosed

### CODE REVIEW

**Review by neiljp — COMMENTED** [review]
@Rohitth007 This is great work :+1: 

This conflicts with the first commit of #1074 - and sort of needs to switch the code/style back? It'd be reasonable to extract the namespace change in that first commit and have it as another commit first in this PR (to set the module-constant precedent to build upon with TAB_WIDTH), and then have this PR's current commits - possibly including the other change from that other PR first commit, but it may be different with this PR code present?

Regarding the second refactor in #1074, as per my comment below I think that would perhaps fit with a vertical-divider refactor commit which would tidy up the inconsistent use of lineboxes? (also enum would be more descriptive than an integer when we're not seeking compatibility with urwid precisely)

File: zulipterminal/ui.py [hunk 1]
```diff
@@ -62,20 +66,20 @@ def middle_column_view(self) -> Any:
         )
 
     def right_column_view(self) -> Any:
+        tab = TabView("❱ USERS ❱")
         self.users_view = RightColumnView(self)
-        return urwid.LineBox(
+        panel = urwid.LineBox(
```
[the remark this answers is not in the document]
Comment (Rohitth007) [reply zulipterminal/ui.py:71]: But a problem arises when we move to the overlayed version of showing tabs. Do we overlay a Column widget with the panel and line widget side by side?

File: zulipterminal/ui.py [hunk 2]
```diff
@@ -45,7 +47,9 @@ def __init__(self, controller: Any) -> None:
         super().__init__(self.main_window())
 
     def left_column_view(self) -> Any:
-        return LeftColumnView(self)
+        tab = TabView("❰ INBOX ❰")
```
[the remark this answers is not in the document]
Comment (Rohitth007) [reply zulipterminal/ui.py:50]: :+1:

File: zulipterminal/ui.py [hunk 3]
```diff
@@ -62,20 +66,20 @@ def middle_column_view(self) -> Any:
...
             title="Users",
             title_attr="column_title",
             tlcorner=COLUMN_TITLE_BAR_LINE,
             tline=COLUMN_TITLE_BAR_LINE,
             trcorner=COLUMN_TITLE_BAR_LINE,
             lline="",
-            blcorner="─",
             rline="",
             bline="",
-            brcorner="",
```
[the remark this answers is not in the document]
Comment (Rohitth007) [reply zulipterminal/ui.py:77]: :+1: I restore it, this woud anyway be removed by the LineBox PR

**Comment by neiljp** [comment]
@Rohitth007 You made a few extra changes compared to the previous version so that needed a little more double-checking, but it looks good :+1: Thanks for tidying this up after the first version, this will make autohide a lot less cryptic - I'm merging it shortly :tada:

Other than a very minor test combination, the only change I made was to use an application-focused name for the symbols.

We could definitely explore refinements to the current design, but that's easier and not as critical as having something present to work with :) We could maybe make them more literally like "tabs" visually, and possibly we don't need "arrows" as such (the current ones also seem offset slightly), and maybe we want to keep the tabs present even when expanded, to show the connection? However, we have a great start point to build from :)

**Pull request merged**